build.sbt

```diff
@@ -229,33 +246,43 @@ lazy val munitScalacheckJVM = munitScalacheck.jvm
 lazy val munitScalacheckJS = munitScalacheck.js
 lazy val munitScalacheckNative = munitScalacheck.native
 
-lazy val tests = crossProject(JSPlatform, JVMPlatform, NativePlatform)
-  .dependsOn(munit, munitScalacheck)
+lazy val testsSettings = List(
+  buildInfoPackage := "munit",
+  buildInfoKeys := Seq[BuildInfoKey](
+    "sourceDirectory" ->
+      baseDirectory.in(ThisBuild).value / "tests" / "shared" / "src" / "main",
+    scalaVersion
+  ),
+  skip in publish := true
+)
+lazy val tests = crossProject(JSPlatform, JVMPlatform)
```

Comment:
Let's keep the tests project unchanged and introduce instead new source directories like `src/test/non-native` for the parts of the cross-build that cats-effect supports. Use `.jvmConfigure(_.dependsOn(munitCatsEffect)).jsConfigure(_.dependsOn(munitCatsEffect))` to skip the native dependency

Reply:
It's also fine to have a `src/test/cats-effect` directory if that simplifies things. I want to avoid multiple test projects because it's easiest to reason about the build when you can run `testsJVM/test` to run all tests.

Reply:
Not sure I completely understand what you mean :)

I currently have created a new directory `munit-cats-effect` at the root of the project.
Would you like to have a new directory in the `tests` folder? Or where do you see this new `non-native` source directory in the project?